philander/stc311.py

```diff
@@ -0,0 +1,665 @@
+        hardware/firmware revision, product/model ID, chip series and alike.
+        For that reason, this function can be used to see,
+        if communication works reliably.
+
+        For more dynamic meta-information see :meth:`getStatus`.
+        
+        The method returns both, an instance of :class:`Info`, carrying
+        the information block as well as an error code, indicating
+        success or failure. The info block shall be evaluated only, if
+        the method returned successfully.
+        Even then, the caller should still evaluate the ``validity``
+        attribute of the returned info block to find out, which of the
+        information is actually valid.
+        
+        :return: The information object and an error code indicating either success or the reason of failure.
+        :rtype: Info, ErrorCode
+        """
+        info = Info()
+        chip_id, err = self.readByteRegister(self.REGISTER.REG_ID)
+        # TODO: not implement yet
+        return info, err
+
+    def getStatus(self, statusID):
+        """Retrieves status data from the device.
+        
+        Typically, this kind of information is more dynamic in that, it
+        changes (much) over time. Usually, it further describes the
+        IC's current shape and condition, such as the availability of
+        new data, the cause of an interrupt or the status of
+        certain hardware functions. Also, secondary measurements such as
+        the die temperature could be subject to status data.
+        
+        For more static meta-information see :meth:`getInfo`.
+        
+        The given ``statusID`` parameter specifies, exactly which status
+        information should be retrieved. Its type and interpretation
+        depends on the implementation.
+        
+        The method returns both, resulting status data and an error code
+        indicating success or failure. The status data should be considered
+        valid only, if the error code indicates a successful execution
+        of this method.
+        
+        The type and interpretation of the status data depends on the
+        specific implementation.
+
+        :param int statusID: Identifies the status information to be retrieved.
+        :return: The status object and an error code indicating either success or the reason of failure.
+        :rtype: Object, ErrorCode
+        """
+        del statusID
+        # TODO: not implemented yet
+        return None, ErrorCode.errNotSupported
+
+    def getBatteryVoltage(self):
+        """Retrieves the battery voltage in milli Volt.
+        
+        :return: A on-negative integer value [mV] or :attr:`Voltage.invalid`\
+        to indicate that this information could not be retrieved.
+        :rtype: Voltage
+        """
+        voltage, err = self.readWordRegister(self.REGISTER.REG_VOLTAGE)
+        ret = self._transferVoltage(voltage) if (err == ErrorCode.errOk) else Voltage.invalid
+        return ret
```

Comment:
Avoid too long lines and function calls within other statements. Let the reviewer see what happens, line by line.